### PR TITLE
Drivetrain ready to add

### PR DESCRIPTION
**Pull request by awurst**
Added commands for drive train. Added drive train subsystem. Added crucial methods for deadzone and getting the axis.

Should be good to go.

### CODE REVIEW

**Comment by SargonRobotics** [comment]
You won't be here to confirm the final merge so I'm going to let someone else add the request.

**Comment by Ubspy** [comment]
We were not aware that you could only issue one pull request per branch. So this one is open again.

File: src/org/usfirst/frc/team2335/robot/OI.java [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+		Axis move = new Axis(controller, Robot.MOVE);
+		Axis rotate = new Axis(controller, Robot.ROTATE);
+		Axis strafe = new Axis(controller, Robot.STRAFE);
+		
+		//drive commands
+		move.whileActive(new Drive());
+		rotate.whileActive(new Turn());
+		strafe.whileActive(new Strafe());
+	}
+
```
Comment (Ubspy) [inline src/org/usfirst/frc/team2335/robot/OI.java:27]: Your comments should be capitalized here. Just for readability, not that important. But always good practice.

File: src/org/usfirst/frc/team2335/robot/Robot.java [hunk 2]
```diff
@@ -20,9 +21,18 @@
 public class Robot extends IterativeRobot
 {	
 	//Constants:
+	public static final double DEADZONE = 0.1;
+	
+	//Axes:
+	public static final int MOVE = 0, ROTATE = 0, SIDESTEP = 0;
+
```
Comment (Ubspy) [inline src/org/usfirst/frc/team2335/robot/Robot.java:20]: Add a TODO: comment saying we need to change the axes to be correct

File: src/org/usfirst/frc/team2335/robot/commands/Drive.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package org.usfirst.frc.team2335.robot.commands;
+
+import org.usfirst.frc.team2335.robot.Robot;
+
+import edu.wpi.first.wpilibj.command.Command;
+
+/**
+ *
+ */
```
Comment (Ubspy) [inline src/org/usfirst/frc/team2335/robot/commands/Drive.java]: Remove this blank comment. Again it's just for neat code.

File: src/org/usfirst/frc/team2335/robot/commands/Strafe.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    // Called once after isFinished returns true
+    protected void end() {
+    	Robot.driveTrain.stopStrafe();
+    }
+
+    // Called when another command which requires one or more of the same
+    // subsystems is scheduled to run
+    protected void interrupted() {
+    }
+}
```
Comment (Ubspy) [inline src/org/usfirst/frc/team2335/robot/commands/Strafe.java:38]: So this file has a lot of readability issues. First, the spacing, second the comments.

Any comments you want to keep or change? If you have none then you should remove all or them.

File: src/org/usfirst/frc/team2335/robot/commands/Turn.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    // Called once after isFinished returns true
+    protected void end() {
+    	Robot.driveTrain.stopDrive();
+    }
+
+    // Called when another command which requires one or more of the same
+    // subsystems is scheduled to run
+    protected void interrupted() {
+    }
+}
```
Comment (Ubspy) [inline src/org/usfirst/frc/team2335/robot/commands/Turn.java:38]: Same thing for this file as the Strafe command, any comments you want to keep. If not the spacing and comments need to be fixed.

File: src/org/usfirst/frc/team2335/robot/triggers/Axis.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	
+	//Axes
+	public static final int MOVE = 1, ROTATE = 0;
+	
+	//Motors
+	public static final int LEFTDRIVE = 1, RIGHTDRIVE = 0, ARM = 8;
+	
+	//Buttons
+	public static final int ARMDOWN = 4, ARMUP = 6;
+
```
Comment (Ubspy) [inline src/org/usfirst/frc/team2335/robot/triggers/Axis.java:18]: There's no reason to use these constants. They're never referenced in any other file. Correct me if I'm wrong though.

File: src/org/usfirst/frc/team2335/robot/triggers/Axis.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+	public Axis(Joystick controller, int axis)
+	{
+		_controller = controller;
+		_axis = axis;
+	}
+	
+	//If the controller value isn't at 0, then it's seen as active
+	//In the OI file it uses .whenActive() or .whileActive()
+	//That means that command will run either once or while this get method is returning true
+    public boolean get()
```
Comment (Ubspy) [inline src/org/usfirst/frc/team2335/robot/triggers/Axis.java:33]: Do we need to keep this comment?

**Review by Ubspy — CHANGES_REQUESTED** [review]
Functionally this looks good, only in the Trigger file is there any optimization issues. But I'm requesting changes for spacing and neat code. Because you guys need good practice and I also have a really bad case of code OCD.

File: src/org/usfirst/frc/team2335/robot/OI.java (hunk 1 above)
Comment (Ubspy) [reply src/org/usfirst/frc/team2335/robot/OI.java:27]: So this issue is fixed, for some reason there are 2 commits that did the same thing.

File: src/org/usfirst/frc/team2335/robot/triggers/Axis.java (hunk 7 above)
Comment (bobfor) [reply src/org/usfirst/frc/team2335/robot/triggers/Axis.java:33]: i don't think we do

Comment (Ubspy) [reply src/org/usfirst/frc/team2335/robot/triggers/Axis.java:33]: For now let's keep it.

File: src/org/usfirst/frc/team2335/robot/commands/Drive.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    // Called once after isFinished returns true
+    protected void end() {
+    	Robot.driveTrain.stopDrive();
+    }
+
+    // Called when another command which requires one or more of the same
+    // subsystems is scheduled to run
+    protected void interrupted() {
+    }
+}
```
Comment (Ubspy) [inline src/org/usfirst/frc/team2335/robot/commands/Drive.java:38]: This file has the same issues as the other ones, pls fix.

**Review by Ubspy — APPROVED** [review]

**Pull request merged**